mlflow/genai/git_versioning/git_info.py

```diff
@@ -0,0 +1,66 @@
+import subprocess
+from dataclasses import dataclass
+
+from typing_extensions import Self
+
+
+@dataclass(kw_only=True)
+class GitInfo:
+    branch: str | None = None
+    commit: str | None = None
+    dirty: bool = False
+
+    @classmethod
+    def from_env(cls) -> Self | None:
+        if not cls._is_git_available():
+            return None
+
+        if not cls._is_in_git_repo():
+            return None
+
+        return cls(
+            branch=cls._get_current_branch(),
+            commit=cls._get_current_commit(),
+            dirty=cls._is_repo_dirty(),
+        )
+
+    @staticmethod
+    def _is_git_available() -> bool:
+        return subprocess.call(["git", "--version"], stderr=subprocess.DEVNULL) == 0
```

Comment:
We could use https://github.com/gitpython-developers/GitPython (which mlflow already depends on), but I didn't want to add another dependency here.

Reply:
I thought `mlflow-skinny` does not depend on `mlflow-skinny` but it actually does. Let me see if it's easy to handle/check git states/info.

Reply:
This is a minor concern, this sounds like `gitpython` will be deprecated or legacy:

<img width="737" height="347" alt="image" src="https://github.com/user-attachments/assets/a52714be-2a1d-4263-9485-6b5856179467" />

Reply:
I've switched to GitPython